packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FGMGroundOverlayController.m

```diff
@@ -34,67 +35,46 @@ - (void)removeGroundOverlay {
   self.groundOverlay.map = nil;
 }
 
-- (void)setConsumeTapEvents:(BOOL)consumes {
-  self.groundOverlay.tappable = consumes;
-}
-
-- (void)setVisible:(BOOL)visible {
-  self.groundOverlay.map = visible ? self.mapView : nil;
-}
-
-- (void)setZIndex:(int)zIndex {
-  self.groundOverlay.zIndex = zIndex;
-}
-
-- (void)setAnchor:(CGPoint)anchor {
-  self.groundOverlay.anchor = anchor;
-}
-
-- (void)setBearing:(CLLocationDirection)bearing {
-  self.groundOverlay.bearing = bearing;
-}
-
-- (void)setTransparency:(float)transparency {
-  float opacity = 1.0 - transparency;
-  self.groundOverlay.opacity = opacity;
-}
-
-- (void)setPositionFromBounds:(GMSCoordinateBounds *)bounds {
-  self.groundOverlay.bounds = bounds;
-}
-
-- (void)setPositionFromCoordinates:(CLLocationCoordinate2D)coordinates {
-  self.groundOverlay.position = coordinates;
-}
-
-- (void)setIcon:(UIImage *)icon {
-  self.groundOverlay.icon = icon;
-}
-
 - (void)updateFromPlatformGroundOverlay:(FGMPlatformGroundOverlay *)groundOverlay
                               registrar:(NSObject<FlutterPluginRegistrar> *)registrar
                             screenScale:(CGFloat)screenScale {
-  [self setConsumeTapEvents:groundOverlay.clickable];
-  [self setZIndex:(int)groundOverlay.zIndex];
-  [self setAnchor:CGPointMake(groundOverlay.anchor.x, groundOverlay.anchor.y)];
-  UIImage *image = FGMIconFromBitmap(groundOverlay.image, registrar, screenScale);
-  [self setIcon:image];
-  [self setBearing:groundOverlay.bearing];
-  [self setTransparency:groundOverlay.transparency];
-  if ([self isCreatedWithBounds]) {
-    [self setPositionFromBounds:[[GMSCoordinateBounds alloc]
-                                    initWithCoordinate:CLLocationCoordinate2DMake(
-                                                           groundOverlay.bounds.northeast.latitude,
-                                                           groundOverlay.bounds.northeast.longitude)
-                                            coordinate:CLLocationCoordinate2DMake(
-                                                           groundOverlay.bounds.southwest.latitude,
-                                                           groundOverlay.bounds.southwest
-                                                               .longitude)]];
+  [FGMGroundOverlayController updateGroundOverlay:self.groundOverlay
+                        fromPlatformGroundOverlay:groundOverlay
+                                      withMapView:self.mapView
+                                        registrar:registrar
+                                      screenScale:screenScale
+                                      usingBounds:self.createdWithBounds];
+}
+
++ (void)updateGroundOverlay:(GMSGroundOverlay *)groundOverlay
```

Comment:
This is the other part of the refactoring I made in every object. Having the class method helps with testing because I can expose this and not need to make a full controller (which for some objects has side effects on the map, which we aren't fully mocking, and it can get complicated). We also had a case where the controller's `init*` method called `updateFrom*`, which violates style, and making the class method allowed trivially fixing that.